app/controllers/users/two_factor_authentication_controller.rb

```diff
@@ -333,12 +334,30 @@ def redirect_url
       if !mobile? && TwoFactorAuthentication::PivCacPolicy.new(current_user).enabled?
         login_two_factor_piv_cac_url
       elsif TwoFactorAuthentication::WebauthnPolicy.new(current_user).enabled?
-        login_two_factor_webauthn_url(webauthn_params)
+        webauthn_url
       elsif TwoFactorAuthentication::AuthAppPolicy.new(current_user).enabled?
         login_two_factor_authenticator_url
       end
     end
 
+    def webauthn_url
+      if webauthn_params[:platform] == true
+        if !device_supports_webauthn_platform?
+          return login_two_factor_options_url
+        end
+      end
+      login_two_factor_webauthn_url(webauthn_params)
+    end
+
+    def device_supports_webauthn_platform?
+      if (!desktop_device? && !BrowserCache.parse(request.user_agent).firefox?) ||
```

Comment:
At first I was wondering if we need to be checking both of these if we're primarily concerned with desktop Firefox, but a re-read of the ticket makes me think that maybe we don't want to check this at all? 

>In this context, "unsupported" primarily means "Firefox", though it would be better if we could check this programmatically, e.g. using [isUserVerifyingPlatformAuthenticatorAvailable](https://developer.mozilla.org/en-US/docs/Web/API/PublicKeyCredential/isUserVerifyingPlatformAuthenticatorAvailable_static).

Since we _are_ checking this programmatically with `isUserVerifyingPlatformAuthenticatorAvailable` and that's what's in `user_session[:platform_authenticator_available]`, maybe we only need to check that value?

Reply:
I think im fine with only checking that value, I know that we have a stricter check for setup similar to this, maybe we can leverage both of the `isWebauthnPasskeySupported` and `isUserVerifyingPlatformAuthenticatorAvailable` instead of desktop device/ firefox check.